impl/src/test/groovy/com/okta/sdk/impl/resource/SessionsTest.groovy

```diff
@@ -0,0 +1,126 @@
+                DefaultSession session = new DefaultSession(dataStore, data)
+                when(dataStore.create((String) eq("/api/v1/sessions"), any(), (Class) eq(Session.class))).thenReturn(session)
+                return dataStore
+            }
+        }
+
+        Session createdSession = client.createSession(new DefaultCreateSessionRequest(null, null)
+                                                                .setSessionToken("101W_juydrDRByB7fUdRyE2JQ"))
+
+        assertThat createdSession.id, equalTo("101W_juydrDRByB7fUdRyE2JQ")
+
+        createdSession.delete()
+        verify(client.dataStore).delete("/api/v1/sessions/101W_juydrDRByB7fUdRyE2JQ")
+    }
+
+    @Test
+    void simpleGetSessionTest() {
+
+        ClientConfiguration clientConfig = new ClientConfiguration()
+        Client client = new DefaultClient(
+                new DefaultClientCredentialsResolver(new TokenClientCredentials("foobar")),
+                new DefaultBaseUrlResolver("https://example.com"),
+                null,
+                new DisabledCacheManager(),
+                clientConfig.getAuthenticationScheme(),
+                clientConfig.getRequestAuthenticatorFactory(),
+                clientConfig.getConnectionTimeout()) {
+            protected InternalDataStore createDataStore(RequestExecutor requestExecutor,
+                                                        BaseUrlResolver baseUrlResolver,
+                                                        ClientCredentialsResolver clientCredentialsResolver,
+                                                        CacheManager cacheManager) {
+                JacksonMapMarshaller mapMarshaller = new JacksonMapMarshaller()
+                Map data = mapMarshaller.unmarshal(this.getClass().getResource( '/stubs/sessions.json' ).text)
+
+                final InternalDataStore dataStore = mock(InternalDataStore)
+                DefaultSession session = new DefaultSession(dataStore, data)
+                when(dataStore.getResource("/api/v1/sessions/101W_juydrDRByB7fUdRyE2JQ", Session)).thenReturn(session)
+                when(dataStore.create((String) eq("/api/v1/sessions/101W_juydrDRByB7fUdRyE2JQ/lifecycle/refresh"), any(), (Class) eq(Session.class))).thenReturn(session)
+                return dataStore
+            }
+        }
+
+        Session session = client.getSession("101W_juydrDRByB7fUdRyE2JQ")
+        assertThat session.id, equalTo("101W_juydrDRByB7fUdRyE2JQ")
+
+        assertThat session.refresh(), notNullValue()
+        verify(client.dataStore).create((String) eq("/api/v1/sessions/101W_juydrDRByB7fUdRyE2JQ/lifecycle/refresh"), any(), (Class) eq(Session.class))
+    }
+
+    @Test
+    void deleteUserSessions() {
+
+        InternalDataStore dataStore = mock(InternalDataStore)
+
+        new DefaultUser(dataStore, [id: "test_user_id"]).endAllSessions()
+        verify(dataStore).delete("/api/v1/users/test_user_id/sessions")
```

Comment:
Do you think we can add more assertions here? Something like GET current session for the user returns invalid.